src/Sentry/SentryHttpMessageHandler.cs

```diff
@@ -69,6 +69,10 @@ internal SentryHttpMessageHandler(IHub? hub, SentryOptions? options, HttpMessage
             $"{method} {url}" // e.g. "GET https://example.com"
             );
         span?.SetExtra(OtelSemanticConventions.AttributeHttpRequestMethod, method);
+        if (!string.IsNullOrWhiteSpace(request.RequestUri?.Host))
+        {
+            span?.SetExtra(OtelSemanticConventions.AttributeServerAddress, request.RequestUri!.Host);
```

Comment:
How about this? This way we don't need the `!` (assumign this compiled)

```suggestion
        if (request.RequestUri is not null && span is not null && !string.IsNullOrWhiteSpace(request.RequestUri.Host))
        {
            span.SetExtra(OtelSemanticConventions.AttributeServerAddress, request.RequestUri.Host);
```